Sources/SimpleHTTP/MultipartForm/MultipartFormData.swift

```diff
@@ -48,16 +48,28 @@ enum Boundary {
 
 struct BodyPart {
     let headers: [Header]
-    let stream: InputStream
     let length: Int
+    let stream: () throws -> InputStream
+
     var hasInitialBoundary = false
     var hasFinalBoundary = false
 
-    init(headers: [Header], stream: InputStream, length: Int) {
+    init(headers: [Header], url: URL, length: Int) {
         self.headers = headers
-        self.stream = stream
+        self.stream = {
+            guard let stream = InputStream(url: url) else {
+                throw MultipartFormData.Error.inputStreamCreationFailed(url)
+            }
+            return stream
+        }
         self.length = length
     }
+
+    init(headers: [Header], data: Data) {
```

Comment:
@rudyfremont 
could you add a common (private) init? It's better to have one in common to avoid any pitfalls. 🙏

Reply:
@pjechris done